File: extract_cost_intelligence.py
import json
import sys

# Read the JSONL file and find Cost Intelligence related messages
with open(r'C:\Users\Admin\.claude\projects\C--Users-Admin-graceful-books\8e3ec30c-6463-48f2-be71-70f2d1e5d9b0.jsonl', 'r', encoding='utf-8') as f:
    messages = []
    for line_num, line in enumerate(f, 1):
        try:
            data = json.loads(line)
            if data.get('type') == 'assistant':
                message = data.get('message', {})
                content = message.get('content', [])

                # Convert content to string for searching
                content_str = json.dumps(content)

                # Check if this message contains Cost Intelligence code
                if any(keyword in content_str for keyword in [
                    'Cost Intelligence',
                    'intelligenceTab',
                    'loadTrendData',
                    'loadVendorIntelData',
                    'generateSmartAlerts',
                    'Scenario Builder',
                    'CPU Trends',
                    'Vendor Intel'
                ]):
                    messages.append({
                        'line': line_num,
                        'uuid': data.get('uuid'),
                        'timestamp': data.get('timestamp'),
                        'content': content
                    })
        except:
            continue

# Print the last 10 messages that mention Cost Intelligence
print(f"Found {len(messages)} messages mentioning Cost Intelligence features\n")
print("=" * 80)

for msg in messages[-10:]:
    print(f"\nLine: {msg['line']}")
    print(f"UUID: {msg['uuid']}")
    print(f"Timestamp: {msg['timestamp']}")
    print("-" * 80)

    for item in msg['content']:
        if item.get('type') == 'tool_use':
            tool_name = item.get('name')
            tool_input = item.get('input', {})

            if tool_name == 'Edit':
                file_path = tool_input.get('file_path', '')
                if 'HistoricalAnalytics' in file_path:
                    print(f"\nTOOL: {tool_name}")
                    print(f"FILE: {file_path}")
                    new_string = tool_input.get('new_string', '')
                    if len(new_string) > 0:
                        print(f"\nNEW_STRING (first 2000 chars):")
                        print(new_string[:2000])
                        if len(new_string) > 2000:
                            print(f"\n... (truncated, total length: {len(new_string)} chars)")

        elif item.get('type') == 'text':
            text = item.get('text', '')
            if any(keyword in text for keyword in ['Cost Intelligence', 'loadTrendData', 'loadVendorIntelData', 'generateSmartAlerts']):
                print(f"\nTEXT:")
                print(text[:1000])
                if len(text) > 1000:
                    print(f"\n... (truncated, total length: {len(text)} chars)")

    print("=" * 80)

File: extract_cost_intelligence.mjs
import { readFileSync, writeFileSync } from 'fs';
import { createInterface } from 'readline';
import { createReadStream } from 'fs';

const conversationFile = 'C:\\Users\\Admin\\.claude\\projects\\C--Users-Admin-graceful-books\\8e3ec30c-6463-48f2-be71-70f2d1e5d9b0.jsonl';

const messages = [];
let lineNum = 0;

const rl = createInterface({
  input: createReadStream(conversationFile),
  crlfDelay: Infinity
});

rl.on('line', (line) => {
  lineNum++;
  try {
    const data = JSON.parse(line);
    if (data.type === 'assistant') {
      const message = data.message || {};
      const content = message.content || [];

      const contentStr = JSON.stringify(content);

      const keywords = [
        'Cost Intelligence',
        'intelligenceTab',
        'loadTrendData',
        'loadVendorIntelData',
        'generateSmartAlerts',
        'Scenario Builder',
        'CPU Trends',
        'Vendor Intel'
      ];

      if (keywords.some(keyword => contentStr.includes(keyword))) {
        messages.push({
          line: lineNum,
          uuid: data.uuid,
          timestamp: data.timestamp,
          content: content
        });
      }
    }
  } catch (e) {
    // Skip malformed lines
  }
});

rl.on('close', () => {
  console.log(`Found ${messages.length} messages mentioning Cost Intelligence features\n`);
  console.log('='.repeat(80));

  // Get the last 15 messages
  const recentMessages = messages.slice(-15);

  let output = '';

  for (const msg of recentMessages) {
    output += `\n\nLine: ${msg.line}\n`;
    output += `UUID: ${msg.uuid}\n`;
    output += `Timestamp: ${msg.timestamp}\n`;
    output += '-'.repeat(80) + '\n';

    for (const item of msg.content) {
      if (item.type === 'tool_use') {
        const toolName = item.name;
        const toolInput = item.input || {};

        if (toolName === 'Edit' || toolName === 'Write') {
          const filePath = toolInput.file_path || '';
          if (filePath.includes('HistoricalAnalytics')) {
            output += `\nTOOL: ${toolName}\n`;
            output += `FILE: ${filePath}\n`;

            const newString = toolInput.new_string || toolInput.content || '';
            if (newString.length > 0) {
              output += `\nCODE CONTENT:\n`;
              output += newString;
              output += '\n';
            }
          }
        }
      } else if (item.type === 'text') {
        const text = item.text || '';
        const keywords = ['Cost Intelligence', 'loadTrendData', 'loadVendorIntelData', 'generateSmartAlerts', 'intelligenceTab'];
        if (keywords.some(keyword => text.includes(keyword))) {
          output += `\nTEXT:\n`;
          output += text;
          output += '\n';
        }
      }
    }

    output += '='.repeat(80) + '\n';
  }

  writeFileSync('cost_intelligence_code_extraction.txt', output);
  console.log('\nExtraction complete! Output saved to cost_intelligence_code_extraction.txt');
  console.log(`Extracted ${recentMessages.length} recent messages`);
});
